src/client-side-metrics/operation-metrics-collector.ts

```diff
@@ -148,9 +154,11 @@ export class OperationMetricsCollector {
       this.state =
         MetricsCollectorState.OPERATION_STARTED_ATTEMPT_NOT_IN_PROGRESS;
       this.attemptCount++;
-      const endTime = new Date();
+      const endTime = hrtime.bigint();
       if (projectId && this.attemptStartTime) {
-        const totalTime = endTime.getTime() - this.attemptStartTime.getTime();
+        const totalTime = Number(
+          (endTime - this.attemptStartTime) / BigInt(1000000)
```

Comment:
nit: it would be good to capture that this is in milliseconds somewhere. Either a comment beside this conversion, or renaming variables to something like `totalMilliseconds` and `operationStartTimeNanos`

Reply:
Also, it looks like this i a float, correct? Is that right?

Reply:
> Also, it looks like this i a float, correct? Is that right?

It looks like it rounds it off for a number type.

![image](https://github.com/user-attachments/assets/26e163ae-273c-42be-b483-66708fe32576)

Reply:
![image](https://github.com/user-attachments/assets/c8bec3b0-c241-4b2f-b8b1-ad78363981bf)

Reply:
And I added comments / variable name changes